Source/DafnyLanguageServer/Workspace/Compilation.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Workspace.ChangeProcessors;
+using SymbolTable = Microsoft.Dafny.LanguageServer.Language.Symbols.SymbolTable;
+using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.Workspace {
+
+  /// <summary>
+  /// Internal representation of a specific version of a Dafny document.
+  ///
+  /// Only one instance should exist of a specific version.
+  /// Asynchronous compilation tasks use this instance to synchronise on
+  ///
+  /// When verification starts, no new instances of Compilation will be created for this version.
+  /// There can be different verification threads that update the state of this object.
+  /// </summary>
+  public class Compilation {
```

Comment:
Let's synchronize the comment that mentions "Dafny document", the variables that are named "document" everywhere and the name of the class that only says "compilation"
I like the name "compilation" because the way you decline it after each phase, but that name is also ambiguous because compilation can also be a process, whereas here you want to return the result.
What about changing `Compilation` to any of these more unambiguous
* `Document` (yes back, to the origins...) --
* `DafnyDocument` (yes back, to the origins...) --
* `CompilationResult`
* `IntermediateDocument`
* `Result`
* `Outcome`

Reply:
Renamed `Compilation` et al. to `Document` et al. , `CompilationManager` to `Compilation`, and `CompilationView` to `IdeState`